src/main/java/net/syscon/elite/service/PrisonToPrisonMoveSchedulingService.java

```diff
@@ -159,16 +159,27 @@ public void cancel(final Long bookingId, final Long scheduledMoveId, final Strin
         checkIsPrison(move);
         checkIsAssociated(bookingId, move);
         checkIsActive(move.getOffenderBooking());
+
+        final var cancelled = eventStatusRepository.findById(EventStatus.CANCELLED).orElseThrow(() -> EntityNotFoundException.withMessage("Event status cancelled not found."));
+
+        if (is(move, cancelled)) {
+            return;
+        }
+
         checkCanCancel(move);
 
-        move.setEventStatus(eventStatusRepository.findById(EventStatus.CANCELLED).orElseThrow(() -> EntityNotFoundException.withMessage("Event status cancelled not found.")));
+        move.setEventStatus(cancelled);
         move.setCancellationReason(transferCancellationReasonRepository.findById(TransferCancellationReason.pk(transferCancellationReasonCode)).orElseThrow(() -> EntityNotFoundException.withMessage("Cancellation reason %s not found.", transferCancellationReasonCode)));
 
         scheduleRepository.save(move);
 
         log.debug("Cancelled scheduled prison to prison move with id {} for offender {}", move.getId(), move.getOffenderBooking().getOffender().getNomsId());
     }
 
+    private boolean is(final OffenderIndividualSchedule move, final EventStatus status) {
+        return move.getEventStatus().equals(status);
```

Comment:
Will the move always have an event status? If these are enums, could use `==` instead?

Reply:
Yes will always have an event status and not an enum.  Reference data.